enumerables.rb

```diff
@@ -0,0 +1,109 @@
+# Implementing  ruby enum methods
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    length.times do |i|
+      yield to_a[i]
+    end
+    self
+  end
+
+  def my_each_with_index
```

Comment:
- [ ]  #my_each_with_index when a block is given when self is a range returns the range itself after calling the given block once for each element in self. The following code should return the range itself. 
```
range.my_each_with_index(&block)
```